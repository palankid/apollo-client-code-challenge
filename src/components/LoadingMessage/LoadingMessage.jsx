import React from 'react';
import { bool, string } from 'prop-types';

import { Spin, Space } from 'antd';

const LoadingMessage = ({ message, visible }) => {
    if (!visible) {
        return null;
    }

    return (
        <div className="loading-message">
            <Space size="middle">
                <Spin size="large" />
                <span className="loading-message__text">
                    { message }
                </span>
            </Space>
        </div>
    );
};

LoadingMessage.propTypes = {
    message: string.isRequired,
    visible: bool
};

LoadingMessage.defaultProps = {
    visible: true
};

export default LoadingMessage;
