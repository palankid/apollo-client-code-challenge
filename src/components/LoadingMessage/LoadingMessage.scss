.loading-message {
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    z-index: 1;

    &__text {
        font-size: 1.5rem;
    }
}
