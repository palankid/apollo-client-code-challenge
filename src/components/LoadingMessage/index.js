import LoadingMessage from './LoadingMessage';
import './LoadingMessage.scss';

export default LoadingMessage;
