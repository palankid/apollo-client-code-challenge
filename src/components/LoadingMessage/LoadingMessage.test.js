import React from 'react';
import { shallow } from 'enzyme';

import LoadingMessage from './LoadingMessage';

describe('LoadingMessage', () => {
  it('should test that component is rendered and not changed unintentionally', () => {
    const wrapper = shallow(
      <LoadingMessage
        message="Loading..."
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should test that component is not rendered', () => {
    const wrapper = shallow(
      <LoadingMessage
        message="Loading..."
        visible={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
