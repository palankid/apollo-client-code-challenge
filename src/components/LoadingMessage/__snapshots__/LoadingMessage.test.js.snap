// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage should test that component is not rendered 1`] = `""`;

exports[`LoadingMessage should test that component is rendered and not changed unintentionally 1`] = `
<div
  className="loading-message"
>
  <Space
    size="middle"
  >
    <Spin
      size="large"
      spinning={true}
      wrapperClassName=""
    />
    <span
      className="loading-message__text"
    >
      Loading...
    </span>
  </Space>
</div>
`;
